app/src/org/commcare/android/analytics/PrefStats.java

```diff
@@ -0,0 +1,90 @@
+package org.commcare.android.analytics;
+
+import android.content.SharedPreferences;
+import android.util.Base64;
+import android.util.Log;
+
+import org.commcare.dalvik.application.CommCareApp;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+/**
+ * Generic logic for storing stats/logs in app preference storage layer.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class PrefStats {
```

Comment:
Ah got it, cool. Should just take the "logs" part out of the class-level comment then.